src/Chainweb/CutDB/Sync.hs

```diff
@@ -103,11 +103,18 @@ syncSession v useOrigin p db logg env pinf = do
         logg @T.Text Debug $ "put cut " <> encodeToText c
 
     receive = do
+        cur <- _cut db
+
         -- Query cut that is at most 1000 blocks ahead
-        h <- _cutHeight <$> _cut db
-        c <- getCut cenv (h + min catchupStepSize (int farAheadThreshold - 1))
-            -- Cf. documentation of 'fastAheadThreshold' for why this bound is
-            -- needed
+        let curHeight = _cutHeight cur
+            curChainCount = length $ view cutMap cur
+            limit = curHeight + min catchupStepSize (int farAheadThreshold * int curChainCount - 1)
```

Comment:
Ideally, we should stop using `CutHeight` wherever possible, because it jumps during graph transitions. However, we can't easily switch to `BlockHeight` here, because the limit is part of the public REST API. So, we would have to extend the API with a query parameter that limits cuts by 'BlockHeight'. We should address this in a separate PR.